### ISSUE TITLE
Plugin does not work with Remote User Tokens

### ISSUE THREAD

**Issue by vbier** [PatrickRoumanoff/nexus-crowd-plugin#32]
I have version 2.9 of the plugin, but this seems to still affect the current version.

When RUT Auth is enabled in Nexus, only local accounts are found. Accounts defined in Crowd are not found. The problem is that CrowdAuthenticatingRealm.getName() does not return the same String as CrowdUserManager.getSource(). 

Look at:
http://grepcode.com/file/repo1.maven.org/maven2/org.sonatype.security/security-system/3.3/org/sonatype/security/DefaultSecuritySystem.java#DefaultSecuritySystem.orderUserManagers%28%29

where the values are put into a map with the realm name as key. A few lines down the map is accessed 
with source as key. If the values do not match, CrowdUserManager is not even used to retrieve user info.
As soon as you change the methods to return the same value, you get exceptions because the rest cache already exists (it looks like the CachingRestClient is now created more than once). As a workaround I simply added a counter to the cache name and everything works. I am not sure what the proper fix for this is, but probably you have an idea.

**Comment by flopma** [comment]
Hello,

Can you publish you security-configuration.xml file please?

So you're using this Crowd plugin v2.9, and what's the version of Nexus?

Thx

**Comment by vbier** [comment]
This is my security-configuration.xml (password has been replaced):

```
<?xml version="1.0"?>
<security-configuration>
  <version>2.0.8</version>
  <anonymousUsername>anonymous</anonymousUsername>
  <anonymousPassword>#####</anonymousPassword>
  <realms>
    <realm>NexusCrowdAuthenticationRealm</realm>
    <realm>XmlAuthenticatingRealm</realm>
    <realm>XmlAuthenticatingRealm</realm>
    <realm>XmlAuthorizingRealm</realm>
    <realm>rutauth-realm</realm>
  </realms>
  <hashIterations>1024</hashIterations>
</security-configuration>
```

The version of Nexus is 2.13.0.

**Comment by flopma** [comment]
The issue is resolved (tested with Nexus 2.14.5). I will release a newer version of the plugin for you (2.9.1)

**Comment by vbier** [comment]
I have just tested your fix and it works. Thanks.
